### PR TITLE
ESM / diary modus

### LINKED ISSUE #123: ESM / diary modus
For the first study, users will be asked to look at their data and reflect on it. In addition, we want to offer them the opportunity to report their reflection (if desired) into the app.

- [x] outline UI/UX concept
- [x] implement text field entry
- [x] store entries in database
- [x] export entries

### PR DESCRIPTION
**Pull request by felixerdy**
closes #123

### CODE REVIEW

**Comment by schrooom** [comment]
There appears to be a merge conflict, possibly created by merging #127.

File: src/app/shared-services/diary/diary.service.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  async exportDiary() {
+    try {
+      const diary = await this.getDiary()
+      const fileData = diary
+        .map((d) => {
+          return `\n${d.date.toISOString()}\n${d.content}\n`
+        })
+        .join(`\n===========\n`)
```
Comment (schrooom) [inline src/app/shared-services/diary/diary.service.ts:61]: What purpose does this joined string serve?

File: src/app/diary/diary-detail/diary-detail.component.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        {
+          text: 'Cancel',
+          role: 'cancel',
+          cssClass: 'secondary',
+          handler: () => {
+            alert.dismiss()
+          },
+        },
+        {
+          text: 'Delete',
```
Comment (schrooom) [inline src/app/diary/diary-detail/diary-detail.component.ts:45]: I'd propose to use `cssClass: 'danger',`  for the Delete-Button to stay consistent with the Delete-Alert of trajectories.

File: src/app/diary/diary.page.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  ngOnInit() {}
+
+  async ionViewWillEnter() {
+    this.diary = await this.diaryService.getDiary()
+  }
+
+  async exportDiary() {
+    try {
+      await this.diaryService.exportDiary()
```
Comment (schrooom) [inline src/app/diary/diary.page.ts:31]: What I'm missing here is the ability to share the exported diary directly via mail where the recipient and subject are pre-filled with the diary file already added as an attachment – if that is even possible?

Comment (schrooom) [reply src/app/diary/diary.page.ts:31]: Might be doable using [this.](https://github.com/katzer/cordova-plugin-email-composer) Haven't looked throughly though.

**Comment by schrooom** [comment]
Looks very good, I like how this turned out! 

Apart from the small merge conflict and stated comments, this is good to go! 👍🏻

File: src/app/shared-services/diary/diary.service.ts (hunk 1 above)
Comment (felixerdy) [reply src/app/shared-services/diary/diary.service.ts:61]: I used the join in order to have a more readable .txt file

Example:
```

2021-12-23T13:16:18.000Z
Hello World

===========

2021-12-23T13:16:30.000Z
this is my second diary entry
```

File: src/app/diary/diary.page.ts (hunk 3 above)
Comment (felixerdy) [reply src/app/diary/diary.page.ts:31]: The [Share plugin](https://capacitorjs.com/docs/v2/apis/share) handles that properly. It requires one additional click but you can share it with any other app as well. 

Regarding pre-filled recipient I thought we could add a modal where one can copy the recipient email address. Unfortunately, that's not possible with the share plugin.

| Share dialog (Android 11) | Email (Gmail) |
|---|---|
| Screenshot_20220106-104641 | Screenshot_20220106-104653 |

File: src/app/diary/diary-detail/diary-detail.component.ts (hunk 2 above)
Comment (felixerdy) [reply src/app/diary/diary-detail/diary-detail.component.ts:45]: resolved in 771aef7

File: src/app/diary/diary.page.ts (hunk 3 above)
Comment (schrooom) [reply src/app/diary/diary.page.ts:31]: Alright. I'd propose to leave it like that for now then. If really needed, we can always add a mechanism for sending a predefined mail later on.

**Comment by schrooom** [comment]
One more thing: I think it would be useful, if the diary-entries were sorted by the given date – within the overview in the app as well as for  the exported file. What do you think?

**Comment by felixerdy** [comment]
> One more thing: I think it would be useful, if the diary-entries were sorted by the given date – within the overview in the app as well as for the exported file. What do you think?

:+1: 
resolved in 29c2ab0

**Pull request merged**